diff --git a/node_modules/@react-native-google-signin/google-signin/android/.classpath b/node_modules/@react-native-google-signin/google-signin/android/.classpath
new file mode 100644
index 0000000..bbe97e5
--- /dev/null
+++ b/node_modules/@react-native-google-signin/google-signin/android/.classpath
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17/"/>
+	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
+	<classpathentry kind="output" path="bin/default"/>
+</classpath>
diff --git a/node_modules/@react-native-google-signin/google-signin/android/.project b/node_modules/@react-native-google-signin/google-signin/android/.project
new file mode 100644
index 0000000..97d573d
--- /dev/null
+++ b/node_modules/@react-native-google-signin/google-signin/android/.project
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-google-signin_google-signin</name>
+	<comment>Project react-native-google-signin_google-signin created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1768953047088</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/@react-native-google-signin/google-signin/android/.settings/org.eclipse.buildship.core.prefs b/node_modules/@react-native-google-signin/google-signin/android/.settings/org.eclipse.buildship.core.prefs
new file mode 100644
index 0000000..68c9fab
--- /dev/null
+++ b/node_modules/@react-native-google-signin/google-signin/android/.settings/org.eclipse.buildship.core.prefs
@@ -0,0 +1,2 @@
+connection.project.dir=../../../../android
+eclipse.preferences.version=1
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/.project b/node_modules/@react-native-google-signin/google-signin/android/bin/.project
new file mode 100644
index 0000000..97d573d
--- /dev/null
+++ b/node_modules/@react-native-google-signin/google-signin/android/bin/.project
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-google-signin_google-signin</name>
+	<comment>Project react-native-google-signin_google-signin created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1768953047088</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/build.gradle b/node_modules/@react-native-google-signin/google-signin/android/bin/build.gradle
new file mode 100644
index 0000000..bed09d6
--- /dev/null
+++ b/node_modules/@react-native-google-signin/google-signin/android/bin/build.gradle
@@ -0,0 +1,79 @@
+buildscript {
+  def kotlin_version = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : project.properties['RNGS_kotlinVersion']
+
+  repositories {
+    google()
+    mavenCentral()
+  }
+
+  dependencies {
+    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${kotlin_version}")
+  }
+}
+
+apply plugin: 'com.android.library'
+apply plugin: 'org.jetbrains.kotlin.android'
+
+def isNewArchitectureEnabled() {
+  // To opt-in for the New Architecture, you can either:
+  // - Set `newArchEnabled` to true inside the `gradle.properties` file
+  // - Invoke gradle with `-newArchEnabled=true`
+  // - Set an environment variable `ORG_GRADLE_PROJECT_newArchEnabled=true`
+  return project.hasProperty("newArchEnabled") && project.newArchEnabled == "true"
+}
+
+if (isNewArchitectureEnabled()) {
+  apply plugin: "com.facebook.react"
+}
+
+def getExtOrIntegerDefault(name) {
+  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties['RNGS_' + name]).toInteger()
+}
+def safeExtGet(prop, fallback) {
+  rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
+android {
+  def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
+  if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
+    namespace "com.reactnativegooglesignin"
+  }
+  compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
+
+  // Used to override the NDK path/version on internal CI or by allowing
+  // users to customize the NDK path/version from their root project (e.g. for M1 support)
+  if (rootProject.hasProperty("ndkPath")) {
+    ndkPath rootProject.ext.ndkPath
+  }
+  if (rootProject.hasProperty("ndkVersion")) {
+    ndkVersion rootProject.ext.ndkVersion
+  }
+
+  defaultConfig {
+    minSdkVersion getExtOrIntegerDefault('minSdkVersion')
+    targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
+    buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
+  }
+
+  sourceSets.main {
+    java {
+      if (isNewArchitectureEnabled()) {
+        srcDirs += 'src/newArch/java'
+      } else {
+        srcDirs += 'src/oldArch/java'
+      }
+    }
+  }
+}
+
+repositories {
+  google()
+  mavenLocal()
+  mavenCentral()
+}
+
+dependencies {
+  //noinspection GradleDynamicVersion
+  implementation "com.facebook.react:react-native:+"  // From node_modules
+  implementation "com.google.android.gms:play-services-auth:${safeExtGet('googlePlayServicesAuthVersion', '21.4.0')}"
+}
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/gradle.properties b/node_modules/@react-native-google-signin/google-signin/android/bin/gradle.properties
new file mode 100644
index 0000000..2dd1feb
--- /dev/null
+++ b/node_modules/@react-native-google-signin/google-signin/android/bin/gradle.properties
@@ -0,0 +1 @@
+RNGS_kotlinVersion=1.8.22
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/AndroidManifest.xml b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/AndroidManifest.xml
new file mode 100644
index 0000000..a4a1494
--- /dev/null
+++ b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/AndroidManifest.xml
@@ -0,0 +1,4 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="com.reactnativegooglesignin">
+
+</manifest>
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/ErrorDto.kt b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/ErrorDto.kt
new file mode 100644
index 0000000..c4b3393
--- /dev/null
+++ b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/ErrorDto.kt
@@ -0,0 +1,42 @@
+package com.reactnativegooglesignin
+
+import com.google.android.gms.auth.api.signin.GoogleSignInStatusCodes
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.UnsupportedApiCallException
+
+class ErrorDto(e: Exception, errCodeFallback: String?) {
+  var code: String? = null
+  var message: String? = null
+
+  init {
+    // Google's exceptions are not very helpful and error codes can be context-sensitive
+    val localizedMessage = e.localizedMessage ?: e.message
+    if (e is ApiException) {
+      val code: Int = e.statusCode
+      // we apply some heuristic to try make the error message more useful
+      val minLengthToBeInsightful = 10
+      val isInsightful =
+          localizedMessage != null && localizedMessage.length > minLengthToBeInsightful
+      // remove code from message, users don't want to see it and it's already present in the code
+      // field
+      val errorDescription: String =
+          if (isInsightful && localizedMessage != null) localizedMessage.replaceFirst("$code: ".toRegex(), "")
+          else GoogleSignInStatusCodes.getStatusCodeString(code)
+
+      // there are several types of cancellations: they can be observed (1) when cancelling sign in
+      // and (2) when calling getAccessToken and dismissing the modal
+      // we treat them as equivalent
+      val isCancelled = code == GoogleSignInStatusCodes.SIGN_IN_CANCELLED || e.status.isCanceled
+      val codeInt = if (isCancelled) GoogleSignInStatusCodes.SIGN_IN_CANCELLED else code
+      this.code = codeInt.toString()
+      this.message = errorDescription
+    } else if (e is UnsupportedApiCallException) {
+      this.code = errCodeFallback
+      this.message =
+          "$localizedMessage Make sure you have the latest version of Google Play Services installed."
+    } else {
+      this.code = errCodeFallback
+      this.message = localizedMessage
+    }
+  }
+}
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/PendingAuthRecovery.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/PendingAuthRecovery.class
new file mode 100644
index 0000000..3b87bf4
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/PendingAuthRecovery.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/PromiseWrapper.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/PromiseWrapper.class
new file mode 100644
index 0000000..a9b7175
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/PromiseWrapper.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninButtonViewManager.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninButtonViewManager.class
new file mode 100644
index 0000000..dc744a0
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninButtonViewManager.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule$AccessTokenRetrievalTask.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule$AccessTokenRetrievalTask.class
new file mode 100644
index 0000000..84cd887
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule$AccessTokenRetrievalTask.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule$RNGoogleSigninActivityEventListener.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule$RNGoogleSigninActivityEventListener.class
new file mode 100644
index 0000000..18974d8
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule$RNGoogleSigninActivityEventListener.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule$TokenClearingTask.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule$TokenClearingTask.class
new file mode 100644
index 0000000..f0a00ee
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule$TokenClearingTask.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule.class
new file mode 100644
index 0000000..5c38810
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninModule.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninPackage.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninPackage.class
new file mode 100644
index 0000000..d16673b
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/RNGoogleSigninPackage.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/SigninButtonEvent.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/SigninButtonEvent.class
new file mode 100644
index 0000000..9e97140
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/SigninButtonEvent.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/Utils.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/Utils.class
new file mode 100644
index 0000000..85e4b40
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/main/java/com/reactnativegooglesignin/Utils.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/newArch/java/com/reactnativegooglesignin/RNGoogleSigninButtonManagerDelegate.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/newArch/java/com/reactnativegooglesignin/RNGoogleSigninButtonManagerDelegate.class
new file mode 100644
index 0000000..3836df1
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/newArch/java/com/reactnativegooglesignin/RNGoogleSigninButtonManagerDelegate.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/newArch/java/com/reactnativegooglesignin/RNGoogleSigninButtonManagerInterface.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/newArch/java/com/reactnativegooglesignin/RNGoogleSigninButtonManagerInterface.class
new file mode 100644
index 0000000..547d75e
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/newArch/java/com/reactnativegooglesignin/RNGoogleSigninButtonManagerInterface.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/NativeGoogleSigninSpec.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/NativeGoogleSigninSpec.class
new file mode 100644
index 0000000..57af473
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/NativeGoogleSigninSpec.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/RNGoogleSigninButtonManagerDelegate.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/RNGoogleSigninButtonManagerDelegate.class
new file mode 100644
index 0000000..01c2929
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/RNGoogleSigninButtonManagerDelegate.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/RNGoogleSigninButtonManagerInterface.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/RNGoogleSigninButtonManagerInterface.class
new file mode 100644
index 0000000..0ffc823
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/RNGoogleSigninButtonManagerInterface.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/RNGoogleSigninButtonManagerInterfaceCopy.class b/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/RNGoogleSigninButtonManagerInterfaceCopy.class
new file mode 100644
index 0000000..191baae
Binary files /dev/null and b/node_modules/@react-native-google-signin/google-signin/android/bin/src/oldArch/java/com.reactnativegooglesignin/RNGoogleSigninButtonManagerInterfaceCopy.class differ
diff --git a/node_modules/@react-native-google-signin/google-signin/plugin/build/withGoogleSignIn.js b/node_modules/@react-native-google-signin/google-signin/plugin/build/withGoogleSignIn.js
index fb2b6a2..8f95392 100644
--- a/node_modules/@react-native-google-signin/google-signin/plugin/build/withGoogleSignIn.js
+++ b/node_modules/@react-native-google-signin/google-signin/plugin/build/withGoogleSignIn.js
@@ -1,7 +1,7 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.withGoogleUrlScheme = void 0;
-const Scheme_1 = require("@expo/config-plugins/build/ios/Scheme");
+// const Scheme_1 = require("@expo/config-plugins/build/ios/Scheme");
 const config_plugins_1 = require("expo/config-plugins");
 const pkg = require('@react-native-google-signin/google-signin/package.json');
 function validateOptions(options) {
@@ -22,6 +22,7 @@ const withGoogleSignInWithoutFirebase = (config, options) => {
 };
 const withGoogleUrlScheme = (config, options) => {
     return (0, config_plugins_1.withInfoPlist)(config, (config) => {
+        const Scheme_1 = require("@expo/config-plugins/build/ios/Scheme");
         config.modResults = (0, Scheme_1.appendScheme)(options.iosUrlScheme, config.modResults);
         return config;
     });
